openedx/adg/lms/applications/models.py

```diff
@@ -48,9 +48,11 @@ def set_is_prerequisite_courses_passed(self):
 
     def set_is_written_application_completed(self):
         """
-        Mark written_application objective as complete i.e set is_written_application_completed to True.
+        Mark written_application objective as complete i.e set is_written_application_completed to True and add current
+        date as submission_date.
         """
         self.is_written_application_completed = True
+        self.submission_date = date.today()
```

Comment:
This submission date is also being set in the ApplicationHub method `submit_application_for_current_date`. So according to the current implementation, the date we set here would be overridden when the application hub is submitted. I know that as per discussion, the application hub logic is to be updated, but the developer might miss out this case. So either you can change the `submit_application_for_current_date` method to not set this date there, or maybe keep a note of this, and add it to the application hub integration story.

Reply:
I have noted it, will add it in the integration story